### ISSUE TITLE
Cannot Area click toggle on stacked area chart (1.8.1)

### ISSUE THREAD

**Issue by wayoutmind** [novus/nvd3#1093]
When updating from 1.7.1 to 1.8.1, I am now encountering the following exception when attempting to click on an area.

```
 Uncaught TypeError: Cannot read property 'y' of undefined
```

This issue exists when running the stacked area example:
https://github.com/novus/nvd3/blob/master/examples/stackedAreaChart.html

**Comment by wayoutmind** [comment]
Any thoughts?

**Comment by liquidpele** [comment]
Sorry, haven't had a chance to look at it yet...  If you want to look at it before I can and send up a fix via pull request then feel free :)

**Comment by jlwerenskjold** [comment]
Hey! I've been working with angular-nvd3 and originally thought that it was the problem with clicking an area of a stacked graph not working but nvd3 is the culprit. To summarize what I posted [here](https://github.com/krispo/angular-nvd3/issues/169), the error that is being thrown is ````TypeError: d.display is undefined```` and is pointed at line 12190 in nvd3: ````.y(function(d) { return d.display.y + d.display.y0 })````

This function expects d to be a point object (I think, not sure what to call it) and is defined on line 12260 of nvd3:
 ````
.on('click', function(d,i) {
                    d3.select(this).classed('hover', false);
                    dispatch.areaClick({
                        point: d,
                        series: d.key,
                        pos: [d3.event.pageX, d3.event.pageY],
                        seriesIndex: d.seriesIndex
                    });
````
Now, for some reason d will sometimes be defined as the bottom graph object of the stacked area graph. For example, in the stacked area graph example [here](https://github.com/novus/nvd3/blob/master/examples/stackedAreaChart.html) d will sometimes be defined as the object:
````Object {key="Consumer Discretionary", values=[77],...```` no matter which area is clicked. In my own graphs the bottom most area on the graph is sometimes defined as d. This is what is causing the error to be thrown. Why this happens I am not sure.

 I tried putting a check in the scatter.y function and that fixed the problem. My scatter.y function is simply:
````
.y(function(d) {
            if(typeof d.display === 'undefined') {}
            else {
              return d.display.y + d.display.y0
            }
          })
````
For the moment this fixes the issue, although getting down to the root of the problem and fixing that should be done as this is just a rudimentary fix.

**Comment by wayoutmind** [comment]
@jlwerenskjold Thanks for this, this workaround does work, but I am seeing (potentially as a side effect) more these exceptions: `Error: Invalid value for <g> attribute transform="translate(NaN,NaN)"`

**Comment by jlwerenskjold** [comment]
@wayoutmind Yeah, I noticed that. It doesn't seem to affect any sort of functionality as it is just a warning and not an actual error, however it is a bit annoying. I haven't been able to find a cause of the warnings but it probably comes from a statement exactly like or similar to:
```wrap.attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');```

**Comment by thiagomurakami** [comment]
Applied this workaround and is doesn't throw exceptions anymore. I'm not sure if it has any side effects.
```javascript
                  .x(function(d){
                  if (d.display !== undefined) return getX(d)
                  else return 0
              })
                .y(function(d) {
                  if (d.display !== undefined) { return d.display.y + d.display.y0; }
                  else return 0
                })
```
